### PR TITLE
[Driver][SYCL]Suppress non-beneficial warning message when building with -fin…

### PR DESCRIPTION
**Pull request by jinge90**
…telfpga -fsycl-link

Signed-off-by: gejin <ge.jin@intel.com>

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @mdtoguchi 
I have updated the tests based on your suggestion, thank you very much!

File: clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// REQUIRES: system-windows
+// RUN:  %clangxx -target x86_64-windows-msvc-pc -fsycl -fintelfpga -fsycl-link -Xshardware %s -### 2>&1 \
+// RUN:  | FileCheck %s
+// RUN:  %clang_cl -target x86_64-windows-msvc-pc -fsycl -fintelfpga -fsycl-link -Xshardware %s -### 2>&1 \
```
Comment (mdtoguchi) [inline clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp:4]:
```suggestion
// RUN:  %clang_cl -target x86_64-pc-windows-msvc -fsycl -fintelfpga -fsycl-link -Xshardware %s -### 2>&1 \
```

Comment (mdtoguchi) [reply clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp:4]: Apologies, I got my syntax messed up when I suggested the change.

File: clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// REQUIRES: system-windows
+// RUN:  %clangxx -target x86_64-windows-msvc-pc -fsycl -fintelfpga -fsycl-link -Xshardware %s -### 2>&1 \
```
Comment (mdtoguchi) [inline clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp:2]:
```suggestion
// RUN:  %clangxx -target x86_64-pc-windows-msvc -fsycl -fintelfpga -fsycl-link -Xshardware %s -### 2>&1 \
```

File: clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp (hunk 1 above)
Comment (jinge90) [reply clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp]: Hi, @mdtoguchi 
Fixed.
Thank you very much.

File: clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp (hunk 2 above)
Comment (jinge90) [reply clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp]: Fixed.

File: clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// REQUIRES: system-windows
+// RUN:  %clangxx -target x86_64-windows-pc-msvc -fsycl -fintelfpga -fsycl-link -Xshardware %s -### 2>&1 \
```
Comment (mdtoguchi) [inline clang/test/Driver/suppress_LNK4221_intelfpga_sycl-link.cpp:2]: Small adjustment needed:  `x86_64-pc-windows-msvc`

**Comment by mdtoguchi** [comment]
Please update commit message to be [Driver][SYCL]

**Comment by DoyleLi** [comment]
[Comment from Xiaodong.Li] Please ignore this comment. It is used to test "on-demand" testing.
/summary:run

File: clang/lib/Driver/ToolChains/MSVC.cpp [hunk 4]
```diff
@@ -322,6 +322,9 @@ void visualstudio::Linker::constructMSVCLibCommand(Compilation &C,
     }
     CmdArgs.push_back(II.getFilename());
   }
+  if (Args.hasArg(options::OPT_fsycl_link_EQ) &&
+      Args.hasArg(options::OPT_fintelfpga))
+    CmdArgs.push_back(C.getArgs().MakeArgString(Twine("/IGNORE:4221")));
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/MSVC.cpp:327]:
```suggestion
    CmdArgs.push_back("/IGNORE:4221");
```

Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/MSVC.cpp:327]: Additional allocation not needed with string literal.

Comment (jinge90) [reply clang/lib/Driver/ToolChains/MSVC.cpp]: Fixed

**Pull request merged**